spec/unit/hathifiles_redirects_verifier_spec.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require "verifier/hathifiles_redirects_verifier"
+require "zinzout"
+
+module PostZephirProcessing
+  RSpec.describe(HathifileRedirectsVerifier) do
+    let(:test_date) { Date.parse("2023-01-01") }
+    let(:verifier) { described_class.new(date: test_date) }
+    let(:redirects_file) { verifier.redirects_file(date: test_date) }
+    let(:redirects_history_file) { verifier.redirects_history_file(date: test_date) }
+    # Including this mess should invalidate either file
+    let(:mess) { "oops, messed up a line!" }
+
+    # Clean dir before each test
+    before(:each) do
+      [ENV["REDIRECTS_DIR"], ENV["REDIRECTS_HISTORY_DIR"]].each do |dir|
+        FileUtils.rm_rf(dir)
+        FileUtils.mkdir_p(dir)
+      end
+    end
+
+    around(:each) do |example|
+      with_test_environment { example.run }
+    end
+
+    def stage_redirects_file
+      FileUtils.cp(fixture("redirects/redirects_202301.txt.gz"), ENV["REDIRECTS_DIR"])
+    end
+
+    def stage_redirects_history_file
+      FileUtils.cp(fixture("redirects/202301.ndj.gz"), ENV["REDIRECTS_HISTORY_DIR"])
+    end
+
+    # Intentionally add mess to an otherwise wellformed file to trigger errors
+    def malform(file)
+      Zinzout.zout(file) do |outfile|
```

Comment:
Could use `Zlib::GzipWriter`, although `Zinzout` is fine.

Reply:
zinzout is gone, replaced with zlib